.ci/install-old-r-packages.R

```diff
@@ -0,0 +1,79 @@
+# [description]
+#
+#    Installs a pinned set of packages that worked together
+#    as of the last R 3.6 release.
+#
+
+.install_packages <- function(packages) {
+    install.packages(  # nolint: undesirable_function
+        pkgs = paste(  # nolint: paste
+            "https://cran.r-project.org/src/contrib/Archive"
+            , packages
+            , sep = "/"
+        )
+        , dependencies = FALSE
+        , lib = Sys.getenv("R_LIBS")
+        , repos = NULL
+    )
+}
+
+# when confronted with a bunch of URLs like this, install.packages() sometimes
+# struggles to determine install order... so install packages in batches here,
+# starting from the root of the dependency graph and working up
+
+# there was only a single release of {praise}, so there is no contrib/Archive URL for it
+install.packages(  # nolint: undesirable_function
+    pkgs = "https://cran.r-project.org/src/contrib/praise_1.0.0.tar.gz"
+    , dependencies = FALSE
+    , lib = Sys.getenv("R_LIBS")
+    , repos = NULL
+)
+
+.install_packages(c(
+    "brio/brio_1.1.4.tar.gz"              # nolint: non_portable_path
```

Comment:
All of these hard-coded versions are the latest for which there's a package at https://cran.r-project.org/src/contrib/Archive. That provides a set of packages that were all working together as of a few days ago.

We shouldn't need to actively manage this list... this should be able to remain untouched (I hope) until we drop R 3 support and delete this script entirely.